ocp_addons_operators_cli/utils/cli_utils.py

```diff
@@ -82,10 +88,24 @@ def run_install_or_uninstall_products(operators, addons, parallel, debug, instal
         for product_action_tuple in addons_action_list + operators_action_list:
             action_func = product_action_tuple[0]
             action_kwargs = product_action_tuple[1]
-            if parallel:
-                futures.append(executor.submit(action_func, **action_kwargs))
-            else:
-                processed_results.append(action_func(**action_kwargs))
+            product_dict = product_action_tuple[2]
+            try:
+                if parallel:
```

Comment:
This need to be converted to be like we have in `openshift-cli-installer`
https://github.com/RedHatQE/openshift-cli-installer/blob/main/openshift_cli_installer/libs/clusters/ocp_clusters.py#L172